skillmap/src/components/AppModal.tsx

```diff
@@ -413,19 +415,103 @@ export class AppModalImpl extends React.Component<AppModalProps, AppModalState>
             </div>
         </Modal>
     }
+
+    renderRewardModal() {
+        const { reward } = this.props;
+
+        if (reward?.type === "certificate") return this.renderCertificateModal(reward);
+        else if (reward?.type === "completion-badge") return this.renderBadgeModal(reward);
+    }
+
+    renderCertificateModal(reward: MapRewardCertificate) {
+        const title = lf("Rewards");
+        const  { mapId, skillMap, activity, hasPendingModals, dispatchNextModal } = this.props;
+
+        const buttons: ModalAction[] = [];
+        buttons.push({
+            label: lf("Open Certificate"),
+
+            onClick: () => {
+                tickEvent("skillmap.reward", { path: mapId, activity: activity!.activityId });
```

Comment:
for `skillmap.reward` vs `skillmap.openCertificate`, is `.reward` intended to be sent for all (it's not currently but would make sense to me)? would it make sense to push this one up to `renderRewardModal`

Reply:
I removed it altogether, I think it's redundant